### PR TITLE
Add missing doc comments

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Review by mikaelm12 — COMMENTED** [review]
Do you also need docs for the `Controllerproperties`, `Filters` `Application` and `Attributes` properties in the `ControllerModel` class?

**Comment by ryanbrandenburg** [comment]
Those items probably need comments too, but I didn't feel confident that I could accurately document them given my knowledge. Instead I documented the areas/methods I was comfortable with, figuring some documentation is better than none.

File: src/Middleware/StaticFiles/src/StaticFileResponseContext.cs [hunk 1]
```diff
@@ -12,11 +12,19 @@ namespace Microsoft.AspNetCore.StaticFiles
     /// </summary>
     public class StaticFileResponseContext
     {
+        /// <summary>
+        /// Constructs the <see cref="StaticFileResponseContext"/>.
+        /// </summary>
         [Obsolete("Use the constructor that passes in the HttpContext and IFileInfo parameters: StaticFileResponseContext(HttpContext context, IFileInfo file)", false)]
```
Comment (Tratcher) [inline src/Middleware/StaticFiles/src/StaticFileResponseContext.cs]: Ah, it's probably time to remove this overload.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**